File: manual_search_in_a_list_array/msiala.ts
function buildArray(target: number[], n: number): string[] {
    const operations: string[] = [];
    let targetIndex = 0;

    for (let num = 1; num <= n; num++) {
        if (targetIndex === target.length) {
            break;
        }

        operations.push("Push");

        if (num === target[targetIndex]) {
            targetIndex++;
        } else {
            operations.push("Pop");
        }
    }

    return operations;
}

File: manual_search_in_a_list_array/msiala.java
import java.util.*;

class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<String> operations = new ArrayList<>();
        int targetIndex = 0;

        for (int num = 1; num <= n; num++) {
            if (targetIndex == target.length) {
                break;
            }

            operations.add("Push");

            if (num == target[targetIndex]) {
                targetIndex++;
            } else {
                operations.add("Pop");
            }
        }

        return operations;
    }
}